tests/e2e/testutils.go

```diff
@@ -437,16 +443,19 @@ func RestartClusterAgent(nodeNames []string) error {
 
 // RunCmdOnNode executes a command from within the given node
 func RunCmdOnNode(cmd string, nodename string) (string, error) {
-	runcmd := "vagrant ssh -c \"" + cmd + "\" " + nodename
+	runcmd := "vagrant ssh " + nodename + " -c \"" + cmd + "\""
 	out, err := RunCommand(runcmd)
 	if err != nil {
-		return out, fmt.Errorf("failed to run command %s on node %s: %v", cmd, nodename, err)
+		return out, fmt.Errorf("failed to run command: %s on node %s: %s, %v", cmd, nodename, out, err)
 	}
 	return out, nil
 }
 
-// RunCommand executes a command on the host
 func RunCommand(cmd string) (string, error) {
+	// we don't use c.env because we embed the command inside a bash shell
+	if k, b := os.LookupEnv("E2E_KUBECONFIG"); b {
+		cmd = "KUBECONFIG=" + k + " " + cmd
+	}
 	c := exec.Command("bash", "-c", cmd)
```

Comment:
I think this should work.

Also, the convention for anything that returns `value, bool` where the bool indicates success, is to call the bool `ok`.

```suggestion
	c := exec.Command("bash", "-c", cmd)
	if kubeconfig, ok := os.LookupEnv("E2E_KUBECONFIG"); ok {
		c.Env = append(os.Environ(), "KUBECONFIG=" + kubeconfig)
	}
```